spec/compiler/parser/parser_spec.cr

```diff
@@ -179,6 +180,7 @@ module Crystal
     # #4815
     assert_syntax_error "def foo!=; end", "unexpected token: !="
     assert_syntax_error "def foo?=(x); end", "unexpected token: ?"
+    assert_syntax_error "def foo?!=(x); end", "unexpected token: !="
```

Comment:
I'm might actually think this shouldn't be a syntax error. It's entirely possible you're unsure of something, but still want to assign a value to it. To avoid mistakes we could enforce adding an additional `!` at the end:
```cr
def foo?!=!(x); end
```

And you could call it like this:
```cr
foo?! = "foo" !